### PR TITLE
Issue/304/move enable apps to separate fragment

### LINKED ISSUE #304: Move enabled apps list to separate Fragment

### PR DESCRIPTION
**Pull request by spuday90**
closes #304

### CODE REVIEW

**Review by adeekshith — CHANGES_REQUESTED** [review]
- [ ] Anywhere on "Enable for apps" should be clickable. As of now, only clicking on the pencil is working.
- [ ] Pencil should center vertically instead of sticking to the top
Screen Shot 2021-06-03 at 12 01 25 AM

**Comment by adeekshith** [comment]
Also, what should I do to change the size of the icon? Looking to reduce their dimension of app "enabled app" icons on MainFragment.

Also make sure app icons have a size boundary. If they are being shown in the same size that the package manager is returning, it can be an issue. They may all not be in the same dimension. Ignore if it is already taken care of.

**Comment by adeekshith** [comment]
It is also not handling all apps disabled case very well.
- [ ] It should not allow all apps to be disabled in the "Enabled Apps" activity. At-least one app should be enabled otherwise should throw a toast.
enabled-apps-all-disabled

- [ ] When all apps are disabled, it still shows WhatsApp icon on main activity. But fixing above should fix this I guess.

**Comment by adeekshith** [comment]
- [ ] (Only on my real device) Main fragment does not get updated after changing enabled apps and coming back. I have to close the app and re-open for it to update (This is where data binding shines 😊).

File: app/src/main/java/com/parishod/watomatic/adapter/SupportedAppsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.parishod.watomatic.R
+import com.parishod.watomatic.model.App
+import com.parishod.watomatic.model.preferences.PreferencesManager
+import com.parishod.watomatic.model.utils.Constants
+import kotlinx.android.synthetic.main.supported_apps_list.view.*
+
+
+class SupportedAppsAdapter(private val listType: Constants.EnabledAppsDisplayType, private var supportedAppsList: ArrayList<App>): RecyclerView.Adapter<SupportedAppsAdapter.AppsViewHolder>() {
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/adapter/SupportedAppsAdapter.kt:16]: Any reason to use `ArrayList` over `List`?

Comment (spuday90) [reply app/src/main/java/com/parishod/watomatic/adapter/SupportedAppsAdapter.kt:16]: no particular reason

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/adapter/SupportedAppsAdapter.kt]: Changed it to `List` dd34942

File: app/src/main/java/com/parishod/watomatic/fragment/MainFragment.java [hunk 2]
```diff
@@ -182,15 +182,14 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         setNumDays();
 
 
-        if(supportedAppsAdapter != null){
-            getEnabledApps();
-            supportedAppsAdapter.updateList(enabledApps);
+        if(supportedAppsAdapter != null) {
+            supportedAppsAdapter.updateList(getEnabledApps());
```
Comment (spuday90) [inline app/src/main/java/com/parishod/watomatic/fragment/MainFragment.java:186]: Oh! This is my mistake, this is not required here it should be in onResume only

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/fragment/MainFragment.java]: Removed 784a975

**Comment by adeekshith** [comment]
@spuday90 lets remove attribution changes from this PR too

**Comment by adeekshith** [comment]
@spuday90 we can ignore showing grayscale icons for disabled apps on main activity for this PR.
We can get this merged as soon as the issues are resolved and work on it in another iteration.

**Comment by adeekshith** [comment]
Great work but as always, I am back with more issues 😃!
I think the case where app was not installed or none of the apps are installed should be handled better.

- [ ] Resolve merge conflicts
- [ ] Apps that are not installed should be shown as disabled on fresh install

**Comment by adeekshith** [comment]
- [ ] Anywhere on "Enable for apps" should be clickable. Some areas are still not clickable.

**Review by adeekshith — APPROVED** [review]

**Comment by adeekshith** [comment]
>     * [ ]  Anywhere on "Enable for apps" should be clickable. Some areas are still not clickable. For example, clicking on the app icons is not working ([previous mention](https://github.com/adeekshith/watomatic/pull/305#pullrequestreview-674876989))

@spuday90 Merging this but can you check this once please. May be file an issue too.

**Pull request merged**